### PR TITLE
chore(tests): test upgrade path from 2.3 to the current PR

### PR DESCRIPTION
**Pull request by mrclay**
- Installs Elgg 2.3
- Upgrades it to this PR via CLI (doesn't include async migrations yet)
- Runs simpletests
- Verifies the home page works via the PHP server

Mostly by @hypeJunction

### CODE REVIEW

**Comment by mrclay** [comment]
@hypeJunction I removed the phpunits on the upgraded codebase. Doesn't seem necessary.

**Comment by mrclay** [comment]
How did I screw this up? https://travis-ci.org/Elgg/Elgg/jobs/261198428#L680

File: engine/classes/Elgg/Config/SettingsMigrator.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	/**
+	 * Migrate dataroot and wwwroot to the settings file (needed in 3.0)
+	 *
+	 * @param Database  $db            Database
+	 * @param string    $settings_path settings.php file path
+	 * @param \stdClass $CONFIG        Elgg's config object
+	 *
+	 * @return void
+	 */
+	public function migrate(Database $db, $settings_path, $CONFIG) {
```
Comment (hypeJunction) [inline engine/classes/Elgg/Config/SettingsMigrator.php]: Does config remain global here?

**Comment by jdalsem** [comment]
great job guys!

File: engine/classes/Elgg/Config/SettingsMigrator.php (hunk 1 above)
Comment (hypeJunction) [inline engine/classes/Elgg/Config/SettingsMigrator.php]: I'd say do not inject `$CONFIG`, return an array of key value pairs instead, and assign the new values up in `Config` class

File: engine/classes/Elgg/Config/SettingsMigrator.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+				$lines = [
+					"",
+					"/**",
+					" * The installation root URL of the site. E.g. https://example.org/elgg/",
+					" *",
+					" * If not provided, this is sniffed from the Symfony Request object",
+					" *",
+					" * @global string \$CONFIG->wwwroot",
+					" */",
+					"\$CONFIG->dataroot = '{$row->url}';",
```
Comment (hypeJunction) [inline engine/classes/Elgg/Config/SettingsMigrator.php]: This should be `wwwroot`

**Comment by hypeJunction** [comment]
@mrclay  Sent you a PR

File: engine/tests/phpunit/Elgg/Config/SettingsMigratorTest.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		$this->settings_path = $settings_path;
+	}
+
+	public function tearDown() {
+		unlink($this->settings_path);
+	}
+
+	public function testDatarootSettingMigration() {
+		if (defined('HHVM_VERSION')) {
+			$this->markTestSkipped();
```
Comment (mrclay) [inline engine/tests/phpunit/Elgg/Config/SettingsMigratorTest.php]: I don't think HHVM supports writing to a file and immediately including it.

**Comment by mrclay** [comment]
@hypeJunction @jdalsem Seems ready!

**Comment by hypeJunction** [comment]
LGTM.

**Pull request merged**